packages/vm/src/imports.rs

```diff
@@ -27,10 +28,17 @@ use crate::sections::encode_sections;
 use crate::serde::to_vec;
 use crate::GasInfo;
 
-const GAS_COST_SECP256K1_VERIFY_SIGNATURE: u64 = 100;
-const GAS_COST_SECP256K1_RECOVER_PUBKEY_SIGNATURE: u64 = 100;
-const GAS_COST_VERIFY_ED25519_SIGNATURE: u64 = 100;
-const GAS_COST_BATCH_VERIFY_ED25519_SIGNATURE: u64 = GAS_COST_VERIFY_ED25519_SIGNATURE / 3;
+// 1000 Cosmos SDK * 100 CosmWasm factor (~154 us in crypto benchmarks)
+const GAS_COST_SECP256K1_VERIFY: u64 = 100000;
+
+// Gas costs relative to secp256k1_verify cost
+const GAS_COST_SECP256K1_RECOVER_PUBKEY: u64 = 105195; // 100_000 * 162 us / 154 us
+const GAS_COST_ED25519_VERIFY: u64 = 40909; // 100_000 * 63 us / 154 us
+
+// Gas costs relative to ed25519_verify cost
+// From https://docs.rs/ed25519-zebra/2.2.0/ed25519_zebra/batch/index.html
+const GAS_COST_ED25519_BATCH_VERIFY: u64 = GAS_COST_ED25519_VERIFY / 2;
+const GAS_COST_ED25519_BATCH_VERIFY_ONE_PUBKEY: u64 = GAS_COST_ED25519_VERIFY / 4;
```

Comment:
Great, thank you.

Could you store all those values in a `GasConfig` struct, implement `Default` with the values above and store it in `Environment`? Then we can easily make the values configurable by the caller later on.

Reply:
Will do.